### PR TITLE
Snaps Home Page

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**

Enables the snaps home page feature in Flask and adds the UI logic to support it. When viewing a snap with a home page, this page will be shown before the current settings page. If the snap has no home page the experience is identical to the previous experience.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1384

## **Manual testing steps**

1. Install the home page example snap from https://metamask.github.io/snaps/test-snaps/latest/
2. Go to the snap home page from the 3-dot menu

## **Screenshots/Recordings**

home-page3

### CODE REVIEW

File: ui/components/app/snaps/snap-home-page/useSnapHome.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        setError(undefined);
+        setLoading(true);
+
+        const newData = await handleSnapRequest({
+          snapId,
+          origin: '',
+          handler: 'onHomePage',
+          request: {
+            jsonrpc: '2.0',
+            method: ' ',
```
Comment (david0xd) [inline ui/components/app/snaps/snap-home-page/useSnapHome.js:22]: Why is the method value just space ` `?

**Review by david0xd — APPROVED** [review]

File: ui/pages/snaps/snap-view/snap-home.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import PropTypes from 'prop-types';
+
+import { SnapHomeRenderer } from '../../../components/app/snaps/snap-home-page/snap-home-renderer';
+import { Box } from '../../../components/component-library';
+
+function SnapHome({ snapId }) {
+  return (
+    <Box>
+      <SnapHomeRenderer snapId={snapId} />
+    </Box>
```
Comment (GuillaumeRx) [inline ui/pages/snaps/snap-view/snap-home.js:11]: Why do we need this component and `SnapHomeRenderer` ? Can't it just be one ?

Comment (FrederikBolding) [reply ui/pages/snaps/snap-view/snap-home.js:11]: `SnapHomeRenderer` may eventually become a page renderer while snap-home is supposed to be just the home page. We may also want to add more logic to snap-home down the line. I think separating them now makes the most sense 🤔

File: ui/components/app/snaps/snap-home-page/useSnapHome.js (hunk 1 above)
Comment (FrederikBolding) [reply ui/components/app/snaps/snap-home-page/useSnapHome.js:22]: Yeah, we don't have a good way to send a request to a handler that does not require a request currently

File: ui/pages/snaps/snap-view/snap-home.js (hunk 2 above)
Comment (GuillaumeRx) [reply ui/pages/snaps/snap-view/snap-home.js:11]: Yes, considering future plans this makes sense !

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**